src/components/NotificationsDrawer/NotificationItem.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import {
+  NotificationDrawerList,
+  NotificationDrawerListItem,
+  NotificationDrawerListItemBody,
+  NotificationDrawerListItemHeader,
+} from '@patternfly/react-core/dist/dynamic/components/NotificationDrawer';
+import { PopoverPosition } from '@patternfly/react-core/dist/dynamic/components/Popover';
+import { Checkbox } from '@patternfly/react-core/dist/dynamic/components/Checkbox';
+import { Label } from '@patternfly/react-core/dist/dynamic/components/Label';
+import { MenuToggle, MenuToggleElement } from '@patternfly/react-core/dist/dynamic/components/MenuToggle';
+import { Dropdown, DropdownItem } from '@patternfly/react-core/dist/dynamic/components/Dropdown';
+import EllipsisVIcon from '@patternfly/react-icons/dist/esm/icons/ellipsis-v-icon';
+import DateFormat from '@redhat-cloud-services/frontend-components/DateFormat';
+import { useDispatch } from 'react-redux';
+import { NotificationData } from '../../redux/store';
+import { markNotificationAsRead, markNotificationAsUnread } from '../../redux/actions';
+
+const NotificationItem = ({ notification }: { notification: NotificationData }) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const dispatch = useDispatch();
+
+  const onCheckboxToggle = () => {
+    dispatch(!notification.read ? markNotificationAsRead(notification.id) : markNotificationAsUnread(notification.id));
+    setIsDropdownOpen(false);
+  };
+
+  const dropdownItems = [<DropdownItem key="read" onClick={onCheckboxToggle}>{`Mark as ${!notification.read ? 'read' : 'unread'}`}</DropdownItem>];
+
+  return (
+    <React.Fragment>
+      <NotificationDrawerList>
+        <NotificationDrawerListItem variant="info" isRead={notification.read}>
+          <NotificationDrawerListItemHeader title={notification.title} srTitle="Info notification:">
+            <Checkbox isChecked={notification.read} onChange={() => onCheckboxToggle()} id="read-checkbox" name="read-checkbox" />
```

Comment:
No need to create anonymous function
```suggestion
            <Checkbox isChecked={notification.read} onChange={onCheckboxToggle} id="read-checkbox" name="read-checkbox" />
```